sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java

```diff
@@ -391,10 +392,30 @@ public void close(final ChannelHandlerContext context, final ChannelPromise prom
         final SslHandler sslHandler = context.pipeline().get(SslHandler.class);
 
         if (sslHandler != null) {
-            // Netty 4.1.36.Final: SslHandler.closeOutbound must be called before closing the pipeline
-            // This ensures that all SSL engine and ByteBuf resources are released
-            // This is something that does not occur in the call to ChannelPipeline.close that follows
-            sslHandler.closeOutbound();
+
+            try {
+                // Netty 4.1.36.Final: SslHandler.closeOutbound must be called before closing the pipeline
+                // This ensures that all SSL engine and ByteBuf resources are released
+                // This is something that does not occur in the call to ChannelPipeline.close that follows
+                sslHandler.closeOutbound();
+            } catch (Exception exception) {
+
+                // Netty will throw the following exception here if the outbound SSL connection has been closed already
+                // javax.net.ssl.SSLException: SSLEngine closed already
+                // Reducing the noise level here because multiple concurrent closes can happen due to race conditions
+                // and there is no harm in this case
+                if (exception instanceof SSLException) {
```

Comment:
Do similar happen for reactor http code paths also?

Reply:
For ReactorNettyClient Kushagra had some discussions with the owner of ReatorNettyClient - https://github.com/reactor/reactor-netty/issues/1165 

My understanding reading through the work item is that this addressed the issues - but I will double-check with @kushagraThapar  when he is back next week